pkg/controller/appdefinition/status.go

```diff
@@ -162,6 +163,46 @@ func JobStatus(req router.Request, resp router.Response) error {
 	return nil
 }
 
+func VolumeStatus(req router.Request, resp router.Response) error {
+	var (
+		app  = req.Object.(*v1.AppInstance)
+		cond = condition.Setter(app, resp, v1.AppInstanceConditionVolumes)
+		pvcs = new(corev1.PersistentVolumeClaimList)
+
+		messages []string
+	)
+
+	if err := req.List(pvcs, &kclient.ListOptions{
+		Namespace: app.Status.Namespace,
+		LabelSelector: klabels.SelectorFromSet(map[string]string{
+			labels.AcornManaged: "true",
+			labels.AcornAppName: app.Name,
+		}),
+	}); err != nil {
+		return err
+	}
+
+	sort.Slice(pvcs.Items, func(i, j int) bool {
+		return pvcs.Items[i].CreationTimestamp.Before(&pvcs.Items[j].CreationTimestamp)
+	})
+
+	for _, pvc := range pvcs.Items {
+		switch pvc.Status.Phase {
+		case corev1.ClaimBound:
+			// No message if the PVC is in phase bound.
+		default:
+			messages = append(messages, fmt.Sprintf("waiting for volume %s to provision and bind", pvc.Labels[labels.AcornVolumeName]))
```

Comment:
@StrongMonkey It is used here. The reason I am adding the label is so that the volume name is displayed to the user instead of any generated name that may have `-bind` or seemingly random characters because of the use of `SafeConcatName`. Essentially do our best to hide any of the implementation details from the user.